tests/bash/comp-tests.bash

```diff
@@ -154,6 +154,26 @@ _completionTests_verifyCompletion "testprog --customComp f" "firstComp forthComp
 _completionTests_verifyCompletion "testprog --customComp=" "firstComp secondComp forthComp" nofile
 _completionTests_verifyCompletion "testprog --customComp=f" "firstComp forthComp" nofile
 
+#################################################
+# Special characters
+#################################################
+if [ "$BASHCOMP_VERSION" = bash2 ]; then
+    orig_bash_comp_no_sort="${BASH_COMP_NO_SORT}"
+    BASH_COMP_NO_SORT=1
+    _completionTests_verifyCompletion "testprog prefix special-chars bash" "bash space bash\\escape bash\\ escaped\\ space bash>redirect bash#comment bash\$var bash|pipe bash;semicolon"
+    _completionTests_verifyCompletion "testprog prefix special-chars bash\\e" "bash\\escape"
+    # TODO: completionTests_verifyCompletion doesn't support testing something
+    # like this.
+    #_completionTests_verifyCompletion "testprog prefix special-chars bash\\ e" "bash\\ escaped\\ space"
```

Comment:
Should we have a test for the `bash space` case provided by the completion logic?

Reply:
Done